Modules/_abc.c

```diff
@@ -588,6 +590,49 @@ _abc__abc_register_impl(PyObject *module, PyObject *self, PyObject *subclass)
     }
     Py_DECREF(impl);
 
+    /* Recursively register the subclass in all ABC bases, to avoid recursive lookups.
+        >>> class Ancestor1(ABC): pass
+        >>> class Ancestor2(Ancestor1): pass
+        >>> class Other: pass
+        >>> Ancestor2.register(Other)  # same result for Ancestor1.register(Other)
+        >>> issubclass(Other, Ancestor2) is True
+        >>> issubclass(Other, Ancestor1) is True
+    */
+    PyObject *mro = PyObject_GetAttrString(self, "__mro__");
+    if (mro == NULL) {
+        return NULL;
+    }
+
+    if (!PyTuple_Check(mro)) {
+        PyErr_SetString(PyExc_TypeError, "__mro__ is not tuple");
+        goto error;
+    }
+
+    for (Py_ssize_t pos = 0; pos < PyTuple_GET_SIZE(mro); pos++) {
+        PyObject *base_class = PyTuple_GET_ITEM(mro, pos);  // borrowed
+        PyObject *base_class_data;
+
+        if (PyObject_GetOptionalAttr(base_class, &_Py_ID(_abc_impl),
+                                 &base_class_data) < 0) {
+            goto error;
+        }
+
+        if (PyErr_Occurred()) {
```

Comment:
`PyObject_GetOptionalAttr` does not set an AttributeError and returns -1 if somthing went wrong, so this should be dead code I think.